pkg/resources/validation/roletemplate/roletemplate.go

```diff
@@ -77,3 +98,35 @@ func (r *roleTemplateValidator) Admit(response *webhook.Response, request *webho
 
 	return r.escalationChecker.ConfirmNoEscalation(response, request, rules, "")
 }
+
+// checkCircularRef looks for a circular ref between this role template and any role template that it inherits
+// for example - template 1 inherits template 2 which inherits template 1. These setups can cause high cpu usage/crashes
+// If a circular ref was found, returns the first template which inherits this role template. Returns nil otherwise.
+// Can return an error if any role template was not found
+func (r *roleTemplateValidator) checkCircularRef(template *v3.RoleTemplate) (*v3.RoleTemplate, error) {
+	seen := make(map[string]struct{}, 0)
+	queue := make([]*v3.RoleTemplate, 0)
```

Comment:
Nit: this could be simplified to just `queue := []*v3.RoleTemplate{template}`.

Reply:
Done - thanks for the suggestion.